### PR TITLE
Deprecate fields that are derived from `containerInfo`

### PR DESCRIPTION
**Pull request by bsideup**

### CODE REVIEW

File: core/src/main/java/org/testcontainers/containers/GenericContainer.java [hunk 1]
```diff
@@ -188,6 +197,10 @@ public GenericContainer(@NonNull final Future<String> image) {
         this.image = image;
     }
 
+    public String getContainerId() {
```
[the remark this answers is not in the document]
Comment (bsideup) [reply core/src/main/java/org/testcontainers/containers/GenericContainer.java]: not deprecated

**Comment by bsideup** [comment]
@kiview yes. I reverted a change to `getContainerId` because we have some usages of `containerId` **before** we query `containerInfo`. 

We will make the field `private` after the deprecation period, so that only `GenericContainer` will be able to use it, and others will have to use `getContainerId` that delegates to `containerInfo?.getId()`

File: core/src/main/java/org/testcontainers/containers/GenericContainer.java [hunk 2]
```diff
@@ -143,12 +143,21 @@
...
      */
     @Setter(AccessLevel.NONE)
+    @Deprecated
     protected String containerId;
 
+    /**
+     * Set during container startup
+     *
+     * @deprecated use {@link GenericContainer#getContainerInfo()}
+     */
```
Comment (rnorth) [inline core/src/main/java/org/testcontainers/containers/GenericContainer.java]: No `@Deprecated` annotation on the field?

Comment (bsideup) [reply core/src/main/java/org/testcontainers/containers/GenericContainer.java]: Good catch, let me add it

Comment (bsideup) [reply core/src/main/java/org/testcontainers/containers/GenericContainer.java]: done 👍

**Comment by rnorth** [comment]
This was released in [1.11.4](https://github.com/testcontainers/testcontainers-java/releases/tag/1.11.4) 🎉

**Pull request merged**